### PR TITLE
removed deprecated Microsoft.Azure packages

### PR DESCRIPTION
**Pull request by acn-sbuad**
## Description
Replaces deprecated Microsoft.Azure dependencies. 

## Related Issue(s)
- #78 
- #77

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
~- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)~
- [ ] All tests run green

### CODE REVIEW

File: src/Storage/Wrappers/KeyVaultClientWrapper.cs [hunk 1]
```diff
@@ -17,31 +14,22 @@ namespace Altinn.Platform.Storage.Wrappers
...
         }
 
         /// <inheritdoc/>
         public async Task<string> GetSecretAsync(string vaultUri, string secretId)
         {
-            // The AzureServiceTokenProvider has its own internal static collection of principals based on the connection string.
-            AzureServiceTokenProvider azureServiceTokenProvider = new AzureServiceTokenProvider(_connectionString);
-            KeyVaultClient keyVaultClient = new KeyVaultClient(new KeyVaultClient.AuthenticationCallback(azureServiceTokenProvider.KeyVaultTokenCallback));
+            // Credentials are set based on environment variables set in Program.cs
+            SecretClient secretClient = new(new Uri(vaultUri), new DefaultAzureCredential());
```
Comment (acn-sbuad) [inline src/Storage/Wrappers/KeyVaultClientWrapper.cs:28]: Could we cache the client per app owner instead of creating a new client for each request?

Comment (SandGrainOne) [reply src/Storage/Wrappers/KeyVaultClientWrapper.cs:28]: We're already caching the value being retrieved.

Comment (acn-sbuad) [reply src/Storage/Wrappers/KeyVaultClientWrapper.cs:28]: ah in the SasTokenProvider, got it!

File: src/Storage/Altinn.Platform.Storage.csproj [hunk 2]
```diff
@@ -10,15 +10,14 @@
...
     <PackageReference Include="Altinn.Common.PEP" Version="1.0.39" />
     <PackageReference Include="JWTCookieAuthentication" Version="2.4.2" />
     <PackageReference Include="Microsoft.ApplicationInsights.AspNetCore" Version="2.21.0" />
     <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="6.0.9" />
     <PackageReference Include="Microsoft.Azure.Cosmos" Version="3.30.1" />
-    <PackageReference Include="Microsoft.Azure.KeyVault" Version="3.0.5" />
-    <PackageReference Include="Microsoft.Azure.KeyVault.Core" Version="3.0.5" />
-    <PackageReference Include="Microsoft.Azure.Services.AppAuthentication" Version="1.6.2" />
     <PackageReference Include="Microsoft.Extensions.Configuration" Version="6.0.1" />
     <PackageReference Include="Microsoft.Extensions.Configuration.AzureKeyVault" Version="3.1.24" />
```
Comment (SandGrainOne) [inline src/Storage/Altinn.Platform.Storage.csproj:22]: The `Microsoft.Extensions.Configuration.AzureKeyVault` also have a dependency to `Microsoft.Azure.KeyVault`. This means it's still included.

Comment (acn-sbuad) [reply src/Storage/Altinn.Platform.Storage.csproj:22]: No need for the package either with the new implementation. Removing it

Comment (acn-sbuad) [reply src/Storage/Altinn.Platform.Storage.csproj]: Removed all unused refs now that I was at it

File: src/Storage/Program.cs [hunk 3]
```diff
@@ -113,22 +113,16 @@ async Task ConnectToKeyVaultAndSetApplicationInsights(ConfigurationManager confi
...
+        SecretClient client = new SecretClient(new Uri(keyVaultSettings.SecretUri), new DefaultAzureCredential());
 
-        string connectionString = $"RunAs=App;AppId={keyVaultSettings.ClientId};" +
-                                  $"TenantId={keyVaultSettings.TenantId};" +
-                                  $"AppKey={keyVaultSettings.ClientSecret}";
-        AzureServiceTokenProvider azureServiceTokenProvider = new AzureServiceTokenProvider(connectionString);
-        KeyVaultClient keyVaultClient = new KeyVaultClient(
-            new KeyVaultClient.AuthenticationCallback(
-                azureServiceTokenProvider.KeyVaultTokenCallback));
-        config.AddAzureKeyVault(
```
Comment (SandGrainOne) [inline src/Storage/Program.cs:124]: We can't remove this line. It's being used to merge values from KeyVault into Configuration. Values important when we connect to CosmosDB.

Comment (acn-sbuad) [reply src/Storage/Program.cs:124]: Nice catch. Locally the connection strings is defined in appsettings, so this didn't fail. Added it in now.

File: src/Storage/Wrappers/KeyVaultClientWrapper.cs [hunk 4]
```diff
@@ -17,31 +14,22 @@ namespace Altinn.Platform.Storage.Wrappers
...
-
         /// <summary>
-        /// Initializes a new instance of the <see cref="KeyVaultClientWrapper"/> class with a client using the credentials from the key vault settings.
+        /// Initializes a new instance of the <see cref="KeyVaultClientWrapper"/> class.
         /// </summary>
-        /// <param name="keyVaultSettings">
-        /// The <see cref="KeyVaultSettings"/> with information about the principal to use when getting secrets from a key vault.
-        /// </param>
-        public KeyVaultClientWrapper(IOptions<KeyVaultSettings> keyVaultSettings)
+        public KeyVaultClientWrapper()
```
Comment (SandGrainOne) [inline src/Storage/Wrappers/KeyVaultClientWrapper.cs:20]: I believe it's unnecessary with an empty constructor.

File: src/Storage/Altinn.Platform.Storage.csproj [hunk 5]
```diff
@@ -10,19 +10,14 @@
...
     <PackageReference Include="Microsoft.ApplicationInsights.AspNetCore" Version="2.21.0" />
-    <PackageReference Include="Microsoft.AspNetCore.Mvc.NewtonsoftJson" Version="6.0.9" />
     <PackageReference Include="Microsoft.Azure.Cosmos" Version="3.30.1" />
-    <PackageReference Include="Microsoft.Azure.KeyVault" Version="3.0.5" />
-    <PackageReference Include="Microsoft.Azure.KeyVault.Core" Version="3.0.5" />
-    <PackageReference Include="Microsoft.Azure.Services.AppAuthentication" Version="1.6.2" />
-    <PackageReference Include="Microsoft.Extensions.Configuration" Version="6.0.1" />
-    <PackageReference Include="Microsoft.Extensions.Configuration.AzureKeyVault" Version="3.1.24" />
-    <PackageReference Include="Microsoft.Extensions.Logging.ApplicationInsights" Version="2.21.0" />
-    <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
```
Comment (SandGrainOne) [inline src/Storage/Altinn.Platform.Storage.csproj:25]: Newtonsoft.Json is still referenced through other packages. This means our own code would still compile without the direct reference. We need to manually verify that Storage have no use for it before removing the reference.

Comment (acn-sbuad) [reply src/Storage/Altinn.Platform.Storage.csproj:25]: reinstated newtonsoft ref

**Review by SandGrainOne — APPROVED** [review]

**Pull request merged**